app/js/scoreChart.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Function to update the chart. Due to how chart.js works, the only effective way to do
+ * this seems to be completely resetting the graph container every time.
+ */
+function updateChart() {
+    document.querySelector('.show-graph').innerHTML = '<canvas id="myChart"></canvas>';
```

Comment:
Did you try using the chart update method? Just want to double-check. Although I suspect that it may not work for this because you're entirely removing and resetting the dataset, would be worth trying if you haven't already. If you have and it didn't work feel free to disregard.

Reply:
As you suspected, graph.js's update functionality is a pain when you're completely replacing the dataset. This isn't a pretty way of doing it, but it's the only way I could find to completely reset the chart on a redraw.